plugins/scaffolder/src/components/fields/EntityPicker/schema.ts

```diff
@@ -13,72 +13,74 @@
 
 /**
  * @public
  */
-export const entityQueryFilterExpressionSchema = z.record(
-  z
+export const entityQueryFilterExpressionSchema = zod.record(
+  zod
     .string()
-    .or(z.object({ exists: z.boolean().optional() }))
-    .or(z.array(z.string())),
+    .or(zod.object({ exists: zod.boolean().optional() }))
+    .or(zod.array(zod.string())),
 );
 
 /**
  * @public
  */
-export const EntityPickerFieldSchema = makeFieldSchemaFromZod(
-  z.string(),
-  z.object({
-    /**
-     * @deprecated Use `catalogFilter` instead.
-     */
-    allowedKinds: z
-      .array(z.string())
-      .optional()
-      .describe(
-        'DEPRECATED: Use `catalogFilter` instead. List of kinds of entities to derive options from',
-      ),
-    defaultKind: z
-      .string()
-      .optional()
-      .describe(
-        'The default entity kind. Options of this kind will not be prefixed.',
-      ),
-    allowArbitraryValues: z
-      .boolean()
-      .optional()
-      .describe('Whether to allow arbitrary user input. Defaults to true'),
-    defaultNamespace: z
-      .union([z.string(), z.literal(false)])
-      .optional()
-      .describe(
-        'The default namespace. Options with this namespace will not be prefixed.',
-      ),
-    catalogFilter: z
-      .array(entityQueryFilterExpressionSchema)
-      .or(entityQueryFilterExpressionSchema)
-      .optional()
-      .describe('List of key-value filter expression for entities'),
-  }),
-);
+export const EntityPickerFieldSchema = makeFieldSchema({
+  output: z => z.string(),
+  uiOptions: z =>
+    z.object({
+      /**
+       * @deprecated Use `catalogFilter` instead.
+       */
+      allowedKinds: z
+        .array(z.string())
+        .optional()
+        .describe(
+          'DEPRECATED: Use `catalogFilter` instead. List of kinds of entities to derive options from',
+        ),
+      defaultKind: z
+        .string()
+        .optional()
+        .describe(
+          'The default entity kind. Options of this kind will not be prefixed.',
+        ),
+      allowArbitraryValues: z
+        .boolean()
+        .optional()
+        .describe('Whether to allow arbitrary user input. Defaults to true'),
+      defaultNamespace: z
+        .union([z.string(), z.literal(false)])
+        .optional()
+        .describe(
+          'The default namespace. Options with this namespace will not be prefixed.',
+        ),
+      catalogFilter: z
+        .array(entityQueryFilterExpressionSchema)
+        .or(entityQueryFilterExpressionSchema)
+        .optional()
+        .describe('List of key-value filter expression for entities'),
+    }),
+});
 
 /**
  * The input props that can be specified under `ui:options` for the
  * `EntityPicker` field extension.
  *
  * @public
  */
-export type EntityPickerUiOptions =
-  typeof EntityPickerFieldSchema.uiOptionsType;
+export type EntityPickerUiOptions = NonNullable<
+  (typeof EntityPickerFieldSchema.TProps.uiSchema)['ui:options']
```

Comment:
Wonder if we should have an equivalent type for this too?

Maybe a `TUiOptions` or something? Can come back to this though.